Gems/Archive/Code/Include/Archive/ArchiveInterfaceStructs.h

```diff
@@ -0,0 +1,276 @@
+        //! The File Path index is used to lookup the location for a file path
+        //! within the archive
+        //! offset = 32
+        AZ::u32 m_tocPathIndexTableUncompressedSize{};
+        //! Uncompressed size of the Table of Contents File Path section
+        //! It contains a blob of FilePaths without any null-termination
+        //! The File Path Index entries are used to look up a file path
+        //! through using the path offset + size entry
+        //! offset = 36
+        AZ::u32 m_tocPathTableUncompressedSize{};
+        //! Uncompressed size of the Table of Contents File Block section
+        //! Contains compressed sizes of individual blocks of a file
+        //! In Archive V1 layout the block size is 2MiB
+        //! offset = 40
+        AZ::u32 m_tocBlockTableSize{};
+
+        //! Threshold value represents the cap on the size a block after it has been
+        //! sent through the compression step to determine if it should be stored compressed
+        //!
+        //! Due to block size defaulting to 2MiB, any blocks that are larger than 2_mib
+        //! after compression will be stored uncompressed.
+        //! So the maximum limit of this value is the Block Size
+        //! offset = 44
+        AZ::u32 m_compressionThreshold{ static_cast<AZ::u32>(ArchiveDefaultBlockSize) };
+
+        //! Stores 32-bit IDS of up to 7 compression algorithms that this archive can use
+        //! offset = 48
+        AZStd::array<Compression::CompressionAlgorithmId, 7> m_compressionAlgorithmsIds;
+
+        //! Offset from the beginning of the File Path index table
+        //! Where the first deleted block is located
+        //! offset = 76
+        AZ::u32 m_firstDeletedFileIndex{ DeletedPathIndexSentinel };
+
+        //! Offset from the beginning of the file block section to the first
+        //! deleted block.
+        //! The first 8 bytes of each deleted block will contain the offset to the next deleted block
+        //! or 0xffff'ffff'ffff'ffff if this is the last deleted block
+        //! offset = 80
+        AZ::u64 m_firstDeletedBlockOffset{ DeletedBlockOffsetSentinel };
+
+        //! total offset = 88
+
+        //! Max FileCount
+        static constexpr AZ::u32 MaxFileCount{(1 << 25) - 1 };
+    };
+
+    static_assert(sizeof(ArchiveHeaderSection) <= ArchiveDefaultBlockAlignment, "Archive Header section should be less than 512 bytes");
+
+    //! Represents an entry of a single file within the Archive
+    struct ArchiveFileMetadataSection
+    {
+        ArchiveFileMetadataSection();
+
+        //! represents the file after it has been uncompressed on disk
+        //! Can represent a file up to 2^35 = 32GiB
+        AZ::u64 m_uncompressedSize : 35; // 35-bits
+        //! compressed files are stored aligned on 512-byte sectors
+        //! Therefore this can represent size up to 35-bits as well
+        AZ::u64 m_compressedSize : 26; // 61-bits
```

Comment:
That because the compressedSize is represented in 512-byte Sectors which is (2^9).
`(2^26) * (2^9) = 2^35`

This is really `m_compressedSizeInSectors`
I'll rename the variable to be accurate